### ISSUE TITLE
login / exit enhancements

### ISSUE THREAD

**Issue by Zbizu** [otland/forgottenserver#870]
It is ok to keep monster attacked when player lost connection or exited without logging out, but:
- a player won't be kicked as long as he targets a monster (he should be kicked few minutes after losing connection even on no-logout tiles to prevent spoofing), it could be done this way: http://www.tibia.com/news/?subtopic=newsarchive&id=2009
- player should have at least 2 seconds after logging in to prepare for fight if he is in unsafe place as here: http://www.tibia.com/news/?subtopic=newsarchive&id=2251
##

**Comment by andersonfaaria** [comment]
@Zbizu did you managed to do this through Lua?

**Comment by Zbizu** [comment]
No. I had exams in that school year (notice that you replied to a comment from 6 years ago), then went to university, haven't been present here for years and I forgot that I wrote that comment.

What I currently think about lua version of that:

login:
player would either need flag unattackable or monsters aggression would have to be suppressed
then after 10 seconds resumed

2 ways of doing it:
1. onLogin -> getSpectators -> addEvent
2. onThink checking last time player logged in (could be very laggy on bigger servers)

exit:
also two ways:
1. (C++) source changes for a stable version that would use functions that were designed for that
2. (Lua) importing curl and getting a glitchy version that would use Player:getIp() (potential downsides of lua version: unattackable player that wasn't dced)
